src/plugins/event-plugins/WebVitalsPlugin.ts

```diff
@@ -25,20 +33,62 @@ export class WebVitalsPlugin extends InternalPlugin {
     // eslint-disable-next-line @typescript-eslint/no-empty-function
     configure(config: any): void {}
 
-    getWebVitalData(webVitalData: Metric, eventType: string): void {
-        const webVitalEvent:
-            | LargestContentfulPaintEvent
-            | FirstInputDelayEvent
-            | CumulativeLayoutShiftEvent = {
+    protected onload(): void {
+        onLCP((metric) => this.handleLCP(metric));
+        onFID((metric) => this.handleFID(metric));
+        onCLS((metric) => this.handleCLS(metric));
+    }
+
+    handleLCP(metric: LCPMetricWithAttribution | Metric) {
```

Comment:
We will never expect any of these handlers to receive just `Metric` because we are using the attribution build of `web-vitals`. However, they left `Metric` in as an argument type, so we should still double check that attributions are attached

Reply:
question: Why do we need to double check that attributions are attached?

Looking at the type definition, it should always be `LCPMetricWithAttribution`, although TypeScript does think it could be `Metric` at this point. If we know it will always be `LCPMetricWithAttribution` then we can simplify this logic.

Reply:
That was my initial thought as well, but I was spooked into double checking because because TypeScript was complaining, like you said. I'm happy to make that change though

Reply:
Removed double check